### PR TITLE
Resolve additional warnings and added files prep CRAN submission

### PR DESCRIPTION
**Pull request by terrytangyuan**
This PR:
* Resolved warnings regarding undocumented `object` arg
* Added NEWS.md and cran-comments.md

Also just submitted a build to windows machine to get some preliminary checks first. I'll post the result here if there are additional issues.

### CODE REVIEW

File: src/interface_r/cran-comments.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+## Test environments
+* local OS X install, R 3.4.3
+* ubuntu 12.04 (on Jenkins), R 3.4.3
+* win-builder (devel and release)
+
+## R CMD check results
+
+0 errors | 0 warnings | 1 note
```
Comment (navdeep-G) [inline src/interface_r/cran-comments.md]: Whats the note?

Comment (terrytangyuan) [reply src/interface_r/cran-comments.md]: Related to new release. See the comment below "This is a new release" that addresses the note. This was auto-generated by devtools. This is assuming https://github.com/h2oai/h2o4gpu/issues/477 is resolved and Erin's vignette PR is merged.

**Comment by mdymczyk** [comment]
Nice 👍 One question regarding versioning - could we make it on par with the Python version? It's in the `make/version.mk` file or does CRAN require us to start with 0.0.1?

**Comment by terrytangyuan** [comment]
CRAN has no requirements on versioning. We can version it whatever. Yea good idea I think we can match versions. @navdeep-G @ledell Any objections?

**Comment by navdeep-G** [comment]
No objections. Me and @mdymczyk recently spoke about this.

**Comment by terrytangyuan** [comment]
Ok sounds good. I'll update the version when we submit to CRAN.

**Pull request merged**